lib/src/parse/parser.dart

```diff
@@ -67,23 +67,29 @@ class Parser {
     if (!scanner.scanChar($dollar)) return false;
     if (!lookingAtIdentifier()) return false;
     identifier();
-    whitespace();
+    whitespace(allowNewlines: false);
     return scanner.scanChar($colon);
   }
 
   // ## Tokens
 
   /// Consumes whitespace, including any comments.
+  ///
+  /// If [allowNewlines] is true, the indented syntax will consume newlines as
+  /// whitespace, in positions when a statement can not end.
   @protected
-  void whitespace() {
+  void whitespace({required bool allowNewlines}) {
```

Comment:
I'd call this `consumeNewlines`, to avoid suggesting that it throws an error when a newline is encountered.